### PR TITLE
Add psql tests

### PR DESCRIPTION
**Pull request by koustuvsinha**
Added tests for Postgresql. @alayek please review

### CODE REVIEW

File: tests/test_modules/psql_test.py [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
-    """ Test that PostgreSQL is properly installed, up and running """
\ No newline at end of file
+    """ Test that PostgreSQL is properly installed, up and running """
+    def checkPresence(self):
+        output = check_output(["psql","--version"]).decode("utf-8").lstrip().rstrip()
+        print(output)
+        return output
+
+    def checkDatabase(self):
+        output = check_output(["psql", "-dfcc_provision", "-c select datname from pg_database"]).decode("utf-8").lstrip().rstrip()
```
Comment (alayek) [inline tests/test_modules/psql_test.py:11]: Wouldn't there be a space between `-d` and `fcc_provision`?

**Comment by alayek** [comment]
Could you copy-paste or share screen shot of the tests? Run `py.test` with `-v` flag. We need to see that these tests are indeed running, and it's not skipping.

File: tests/test_modules/psql_test.py (hunk 1 above)
Comment (koustuvsinha) [inline tests/test_modules/psql_test.py:11]: I gave a space, but it gave me the error : `psql: FATAL:  database " fcc_provision" does not exist`... maybe internally the library is splitting directly after `-d` ?

**Comment by koustuvsinha** [comment]
@alayek 

image

**Comment by alayek** [comment]
Please remove the `test_runner.py`. I don't think we need that at all. Try to run the tests without the `test_runner`. This would make writing tests even easier. All you need to do to add tests would be to just add a Python file with a class in it. It's test methods would be executed.

File: tests/test_modules/psql_test.py [hunk 2]
```diff
@@ -1,4 +1,21 @@
 from subprocess import call, check_output
+import psycopg2
```
Comment (alayek) [inline tests/test_modules/psql_test.py]: This is not being used I suppose. In that case, please remove this.

**Comment by koustuvsinha** [comment]
@alayek removed test_runner and implemented changes. for postgresql, should we need more tests?

**Comment by alayek** [comment]
We need more tests, but only Django centric. Maybe we would add that tonight. We would build a Django project and build a simple model layer to test CRUD functionality. But that would be in a separate file and different class.

For now, this would do.

# LGTM :+1:

File: tests/test_modules/psql_test.py (hunk 1 above)
Comment (alayek) [inline tests/test_modules/psql_test.py:11]: Perhaps. I will check.

**Comment by koustuvsinha** [comment]
@alayek updated

**Comment by alayek** [comment]
@koustuvsinha thanks! I will test on my system once and merge :100:

**Comment by BKinahan** [comment]
Tested on Mac, LGTM.

```
============================= test session starts ==============================
platform linux -- Python 3.4.3, pytest-2.9.1, py-1.4.31, pluggy-0.3.1 -- /home/vagrant/.virtualenvs/fcc-py3/bin/python3
cachedir: .cache
rootdir: /vagrant/tests, inifile:
collected 5 items

test_modules/git_test.py::TestGit::test_git_version PASSED
test_modules/os_test.py::TestOs::test_uname_kernel PASSED
test_modules/os_test.py::TestOs::test_uname_os PASSED
test_modules/psql_test.py::TestPsql::testPsqlConnection PASSED
test_modules/psql_test.py::TestPsql::testPsqlDatabase PASSED

=========================== 5 passed in 0.19 seconds ===========================
```

**Comment by byteknacker** [comment]
@BKinahan thanks for testing this PR. Would you mind if I invite you to our slack channel where we develop the application together and communicate about this repo?

**Comment by BKinahan** [comment]
@byteknacker Sure, can you invite through github or will you need my e-mail address? I don't have the most experience but I'd be happy to run tests and help however I can.

**Comment by byteknacker** [comment]
@BKinahan please follow this link to sign up. You just need to put your email and name. We keep them confidential. https://johnwu.typeform.com/to/ifhehu

**Pull request merged**